src/main/kotlin/calculator/Operation.kt

```diff
@@ -0,0 +1,44 @@
+package calculator
+
+import calculator.exceptions.NumberSeparationException
+
+enum class Operation(
+    private val sign: String,
+) {
+    ADD("+") {
+        override fun apply(
+            a: Double,
+            b: Double,
+        ) = a + b
+    },
+    SUBTRACT("-") {
+        override fun apply(
+            a: Double,
+            b: Double,
+        ) = a - b
+    },
+    MULTIPLY("*") {
+        override fun apply(
+            a: Double,
+            b: Double,
+        ) = a * b
+    },
+    DIVIDE("/") {
+        override fun apply(
+            a: Double,
+            b: Double,
+        ) = a / b
+    },
+    ;
+
+    abstract fun apply(
+        a: Double,
+        b: Double,
+    ): Double
+
+    companion object {
+        fun parseOperation(inputOperation: String) =
```

Comment:
These are called factory functions, and they're typically named something like `from()` or `of()`. I hope the article below helps:
https://kt.academy/article/ek-factory-functions